src/components/CreateVmWizard/helpers.js

```diff
@@ -15,17 +15,55 @@ const handleClusterIdChange = (clusterId, blankTemplateId, { templates, clusters
 
     changes.dataCenterId = clusters.getIn([clusterId, 'dataCenterId'])
     changes.clusterId = clusterId
-    changes.provisionSource =
-      (templateList.length === 1 && templateList[0].id === blankTemplateId)
-        ? 'iso'
-        : 'template'
-    changes.isoImage = undefined
-    changes.templateId = templateList.length === 1 ? blankTemplateId : undefined
+
+    if (templateList.length > 1) {
```

Comment:
Mutable `changes` object requires checking the whole method to determine what is returned.
For me it would be easer to return the complete object in each branch (and flatten the branch structure too)
i.e.
```
if (templateList.length > 2) {
   return {
        clusterId,
        dataCenterId,
        ...handleProvisionSourceChange(...)  
   }  
} else if (templateList.length == 2) {
....
} else if (templateList.length === 1 && templateList[0].id === blankTemplateId) {
  ...
  if (isoList.length === 1) {
     return { ...}
  else {
    return { ...}
  }
}
else {
 .....
}
```